File: Java Script - Introduction to Programming/9.9. Functions/14. Numbers to Words.js
function letterize([...arg]) {

    for (let i = 1; i < arg.length; i++) {
        let number = +arg[i]
        if (number > 999) console.log('too large')
        if (number < -999) console.log('too small')
        else if (Math.abs(number) > 99 && number < 1000) {
            if (number < 0) {
                console.log('minus ' + numberToText(number, 'negative'))
            } else {
                console.log(numberToText(number))
            }

        }
    }

    function numberToText(number, negative) {
        const [digit1, digit2, digit3] = String(Math.abs(number)).split('')

        const arrOne = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine',
            'ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen',
            'seventeen', 'eighteen', 'nineteen'
        ]

        const arrTwo = ['', '', 'twenty', 'thirty', 'forty', 'fifty', 'sixty',
            'seventy', 'eighty', 'ninety',
        ]

        let result = ''
        result += arrOne[digit1] + '-hundred and ' + ''

        num = Number(digit2 + digit3)
        if(num === 0)return arrOne[digit1] + '-hundred'
    
        if (num >= 0 && num < 10) {
            result += arrOne[digit3]
        } else if (num >= 0 && num <= 19) {
            result += arrOne[digit2 + digit3]
        } else if (num >= 20 && num <= 100) {
            if (digit3 == 0) {
                result += arrTwo[digit2]
            } else {
                result += arrTwo[digit2] + ' ' + arrOne[digit3]
            }
        }

        return result
    }
}


// letterize([2, 500, 123])
// letterize([3, 999, -420, 1020])
// letterize([4, 311, 418, 509, -9945])

File: Pyrhon - Introduction to Programming/9.9. Functions/14. Numbers to Words.py
def letterize(num):

    d = { 0 : 'zero', 1 : 'one', 2 : 'two', 3 : 'three', 4 : 'four', 5 : 'five',
          6 : 'six', 7 : 'seven', 8 : 'eight', 9 : 'nine', 10 : 'ten',
          11 : 'eleven', 12 : 'twelve', 13 : 'thirteen', 14 : 'fourteen',
          15 : 'fifteen', 16 : 'sixteen', 17 : 'seventeen', 18 : 'eighteen',
          19 : 'nineteen', 20 : 'twenty',
          30 : 'thirty', 40 : 'forty', 50 : 'fifty', 60 : 'sixty',
          70 : 'seventy', 80 : 'eighty', 90 : 'ninety' }

    k = 1000
    m = k * 1000

    negative = False
    if num > 999: return 'too large'

    if num < 0:
        if num < -999:return 'too small'
        else:
            negative = True
            num = abs(num)


    if num < 20:
        if negative:
            return 'minus ' + d[num]
        return d[num]

    if num < 100:
        if negative:
            if num % 10 == 0: return 'minus ' + d[num]
        if num % 10 == 0: return d[num]

        else:
            if negative: return 'minus ' + d[num // 10 * 10] + ' ' + d[num % 10]
            return d[num // 10 * 10] + ' ' + d[num % 10]

    if num < k:
        if negative:
            if num % 100 == 0: return 'minus ' + d[num // 100] + '-hundred'
        if num % 100 == 0: return d[num // 100] + '-hundred'
        else:
            if negative:
                return 'minus ' +  d[num // 100] + '-hundred and ' + letterize(num % 100)
            return d[num // 100] + '-hundred and ' + letterize(num % 100)



loop = int(input())

for _ in range(loop):
    current_num = int(input())
    if len(str(current_num)) < 3:
        continue
    print(letterize(current_num))


'''
3
999
-420
1020
--------
1
-420
'''
